packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1831,11 +1831,11 @@ class _TimePickerDialogState extends State<_TimePickerDialog> {
           child: ButtonBar(
```

Comment:
Could we eliminate the ButtonBar here?

Reply:
We can, but then we would lose some of the padding around the buttons that the `ButtonBar` provides. Might trip up some golden tests.

Reply:
I think you can get the same effect with
```dart
Container(
  alignment: Alignment.center,
  padding: EdgeInsets.symmetric(horizontal: 8),
  constraints: const BoxConstraints(minHeight: 52.0),
  child: Row(
    mainAxisAlignment: MainAxisAlignment.end,
    children: <Widget>[
      TextButton(
        onPressed: _handleCancel,
        child: Text(widget.cancelText ?? localizations.cancelButtonLabel),
      ),
      const SizedBox(width: 8),
      TextButton(
        onPressed: _handleOk,
        child: Text(widget.confirmText ?? localizations.okButtonLabel),
      ),
    ],
  ),
)
```

Reply:
There's special behavior in ButtonBar that puts the buttons in a column if they don't fit in a row (good for i18n cases or custom button text), so I'd say we keep the button bar until we have a good replacement for that

Reply:
Is that really an issue with the time picker? Can the picker become narrow enough to require making a column out of the OK/Cancel buttons while otherwise remaining usable?

Reply:
I think it may be possible in other locales (german perhaps?)

Reply:
OK, with me to leave this as-is for the moment. I'm going to write a row-unless-theres-no-room-then-a-column that we can replace ButtonBar with. Until then, this is good.